File: june14/july10_Rec6/permutations.cpp
#include<iostream>
#include<vector>
#include<string>

using namespace std;

//objects on levels and boxes on options
int count=0;
void printPerm(vector<bool>& boxes,int co,int to,string ans)//current object and total object
{
    if(co == to)
    {
        cout << ans << endl;
        count++;
        return;
    }
    for(int i=0;i<boxes.size();i++)
    {
        if(boxes[i] == false)
        {
            boxes[i] = true;
            printPerm(boxes,co+1,to,ans+"O"+to_string(co+1)+"B"+to_string(i)+" ");
            boxes[i] = false;
        }
    }
}

int main(int argc,char** argv)
{
    vector<bool> boxes (4,0); // size , default value
    int to=2;
    printPerm(boxes,0,to,"");
    cout << count << endl;
}

File: june14/july10_Rec6/permutations.java
import java.util.*;
public class permutations{

    static void printPerm(boolean[] boxes,int co,int to,String ans)//current object and total object
    {
        if(co == to)
        {
            System.out.println(ans);
            return;
        }
        for(int i=0;i<boxes.length;i++)
        {
            if(boxes[i] == false)
            {
                boxes[i] = true;
                printPerm(boxes,co+1,to,ans+"O"+(co+1)+"B"+i+" ");
                boxes[i] = false;
            }
        }
    }
    public static void main(String[] args) {
        boolean[] boxes = new boolean[4];
        printPerm(boxes, 0, 2, "");
    }
}